### PR TITLE
Fix Group Message member count on GraphQL scenario

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
We were not fetching the profiles on the GraphQL scenario since we already had the pretty display name. This caused the member count to be always 0.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50714

#### Release Note
```release-note
Fix Group Message member count showing as 0 on GraphQL enabled instances.
```

### CODE REVIEW

File: app/actions/remote/entry/gql_common.ts [hunk 1]
```diff
@@ -98,6 +102,16 @@ export async function deferredAppEntryGraphQLActions(
     updateCanJoinTeams(serverUrl);
     updateAllUsersSince(serverUrl, since);
 
+    // defer sidebar GM profiles
+    setTimeout(async () => {
+        const directChannels = chData?.channels?.filter((v) => v?.type === General.GM_CHANNEL);
+        const channelsToFetchProfiles = new Set<Channel>(directChannels);
+        if (channelsToFetchProfiles.size) {
+            const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], config.LockTeammateNameDisplay, config.TeammateNameDisplay, license);
+            fetchMissingDirectChannelsInfo(serverUrl, Array.from(channelsToFetchProfiles), currentUserLocale, teammateDisplayNameSetting, currentUserId);
```
Comment (enahum) [inline app/actions/remote/entry/gql_common.ts:111]: does it make sense to call `fetchProfilesInGroupChannels` instead of this?

Comment (larkox) [reply app/actions/remote/entry/gql_common.ts:111]: You are completely right. I was using fetchMissingDirectChannelsInfo to make sure I was not missing anything, but for GMs the only thing important in this case is fetching the profiles.

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/actions/remote/entry/gql_common.ts [hunk 2]
```diff
@@ -98,6 +100,20 @@ export async function deferredAppEntryGraphQLActions(
...
 
+    // defer sidebar GM profiles
+    setTimeout(async () => {
+        const gmIds = chData?.channels?.reduce<string[]>((acc, v) => {
+            if (v?.type === General.GM_CHANNEL) {
+                acc.push(v.id);
+            }
+            return acc;
+        }, []);
+        const channelsToFetchProfiles = new Set<string>(gmIds);
```
Comment (enahum) [inline app/actions/remote/entry/gql_common.ts:111]:
```suggestion
        const channelsToFetchProfiles = chData?.channels?.reduce<Set<string>>((acc, v) => {
            if (v?.type === General.GM_CHANNEL) {
                acc.add(v.id);
            }
            return acc;
        }, new Set<string>());
```

I think this is better, no?

**Review by enahum — CHANGES_REQUESTED** [review]

**Review by enahum — APPROVED** [review]

File: app/actions/remote/entry/gql_common.ts [hunk 3]
```diff
@@ -98,6 +100,19 @@ export async function deferredAppEntryGraphQLActions(
     updateCanJoinTeams(serverUrl);
     updateAllUsersSince(serverUrl, since);
 
+    // defer sidebar GM profiles
```
Comment (M-ZubairAhmed) [inline app/actions/remote/entry/gql_common.ts:103]: didnt knew mobile used this deferred request approach, interesting

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**